### PR TITLE
feat: optimized rejected sampler for Eagle3.

### PR DESCRIPTION
**Pull request by RobbieLeung**

### CODE REVIEW

**Review by XuZhang99 — APPROVED** [review]

File: xllm/core/framework/sampling/sampler.cpp [hunk 1]
```diff
@@ -72,7 +72,7 @@ SampleOutput Sampler::forward(torch::Tensor& logits,
     auto greedy = greedy_sample(probs);
     samples = torch::where(params.do_sample, random, greedy);
   }
-  output.probs = probs;
+  output.probs = probs.to(logits.dtype());
```
Comment (Clement-Wang26) [inline xllm/core/framework/sampling/sampler.cpp:75]: probs is float32 and logits is bf16, so should we cast logits to float32 instead?

**Review by Clement-Wang26 — APPROVED** [review]
